ros_ws/src/car_control/src/dms_controller.cpp

```diff
@@ -0,0 +1,73 @@
+#include "dms_controller.h"
+
+/**
+ * Class constructor that sets up a subscriber that listens for dms messages.
+ * */
+DMSController::DMSController()
+{
+    this->m_dms_subscriber =
+        this->m_node_handle.subscribe<std_msgs::Int64>(TOPIC_DMS, 1, &DMSController::dmsCallback, this);
+
+    this->m_command_pulisher = this->m_node_handle.advertise<std_msgs::String>(TOPIC_COMMAND, 1);
+}
+
+void DMSController::checkDMS()
+{
+    struct timeval time_struct;
+    gettimeofday(&time_struct, NULL);
+    long int timestamp = time_struct.tv_sec * 1000 + time_struct.tv_usec / 1000;
+
+    //
+    if (m_running)
+    {
+        // last dms message is outdated
+        if (m_last_dms_message_received + DMS_EXPIRATION < timestamp)
+        {
+
+            m_running = false;
+            std_msgs::String command_message;
+            command_message.data = "stop";
+            this->m_command_pulisher.publish(command_message);
+        }
+        // dms works fine
+        else
+        {
+        }
+    }
+    if (!m_running)
```

Comment:
is it intended that this codes runs if `m_running` was at the beginning of the function call `true` and was set to `false` in line 27? If yes, please add an comment explaining what the block above (the `if(m_running)` one) does and what this block does. If no, might es well make a simple `else` out of the `if`